docs/data.fetch.md

```diff
@@ -0,0 +1,82 @@
+# qgrid filtering interface
+
+Filtering is tightly coupled with ordering, paging and virtualization.
+
+Data fetch process is going to be a pipline of ordered operations:
+
+ 1. **Fetch** - returns a promise which returns data.
+ 2. **Filter** (optional) - filters data according to `match` function. Gets data from `Fetch` stage.
+ 3. **Order** (optional) - orders data according to `comparator` function provided by user. Gets data from `Filter` stage. 
+ 4. **Page** (optional) - defines current page and page size  
+ 5. **Virtualize** (optional) - defines number of rendered items.
+
+Every operation gets data from previous stage and handles it properly.
+If user wants to perform some of this actions on client-side, he modifies pipeline by adding a function to corresponding model.
+
+## Common cases
+1. All data is filtered and ordered on server-side.
+```javascript
+gridModel
+    .data({
+        fetch: (model) => {
+            return $http({
```

Comment:
1. Yes, fetch returns promise.
2. It depends on user-defined logic. If pagination is performed by server, then yes.
3. All this pipeline should be triggered when there are any changes in filter, order or page. Also we need to expose callback for user, `invalidate` function to trigger it immediately.
4.